src/mesh/mesh_communication.C

```diff
@@ -1001,9 +1001,6 @@ void MeshCommunication::gather (const processor_id_type, DistributedMesh &) cons
 // ------------------------------------------------------------
 void MeshCommunication::gather (const processor_id_type root_id, DistributedMesh & mesh) const
 {
-  // The mesh should know it's about to be serialized
-  libmesh_assert (mesh.is_serial());
```

Comment:
It caused that assert to fail because the assert is just wrong.  This `gather()` function can take a processor ID to gather to.  In that case the mesh is NOT serialized (at least not everywhere).